sql/src/main/java/io/crate/executor/transport/TransportSchemaUpdateAction.java

```diff
@@ -0,0 +1,194 @@
+import org.elasticsearch.action.admin.indices.mapping.put.PutMappingRequest;
+import org.elasticsearch.action.admin.indices.mapping.put.PutMappingResponse;
+import org.elasticsearch.action.support.ActionFilters;
+import org.elasticsearch.action.support.master.TransportMasterNodeAction;
+import org.elasticsearch.client.node.NodeClient;
+import org.elasticsearch.cluster.ClusterState;
+import org.elasticsearch.cluster.ClusterStateUpdateTask;
+import org.elasticsearch.cluster.block.ClusterBlockException;
+import org.elasticsearch.cluster.metadata.IndexNameExpressionResolver;
+import org.elasticsearch.cluster.metadata.IndexTemplateMetaData;
+import org.elasticsearch.cluster.metadata.MetaData;
+import org.elasticsearch.cluster.service.ClusterService;
+import org.elasticsearch.common.Priority;
+import org.elasticsearch.common.compress.CompressedXContent;
+import org.elasticsearch.common.inject.Inject;
+import org.elasticsearch.common.inject.Singleton;
+import org.elasticsearch.common.settings.Settings;
+import org.elasticsearch.common.unit.TimeValue;
+import org.elasticsearch.common.xcontent.NamedXContentRegistry;
+import org.elasticsearch.common.xcontent.XContentBuilder;
+import org.elasticsearch.common.xcontent.XContentHelper;
+import org.elasticsearch.common.xcontent.XContentParser;
+import org.elasticsearch.common.xcontent.json.JsonXContent;
+import org.elasticsearch.index.Index;
+import org.elasticsearch.threadpool.ThreadPool;
+import org.elasticsearch.transport.TransportService;
+
+import java.util.Map;
+import java.util.concurrent.CompletableFuture;
+
+import static org.elasticsearch.index.mapper.MapperService.parseMapping;
+
+@Singleton
+public class TransportSchemaUpdateAction extends TransportMasterNodeAction<SchemaUpdateRequest, SchemaUpdateResponse> {
+
+    private final NodeClient nodeClient;
+    private final NamedXContentRegistry xContentRegistry;
+
+    @Inject
+    public TransportSchemaUpdateAction(Settings settings,
+                                       TransportService transportService,
+                                       ClusterService clusterService,
+                                       ThreadPool threadPool,
+                                       ActionFilters actionFilters,
+                                       IndexNameExpressionResolver indexNameExpressionResolver,
+                                       NodeClient nodeClient,
+                                       NamedXContentRegistry xContentRegistry) {
+        super(settings,
+            "crate/sql/ddl/schema_update",
+            transportService,
+            clusterService,
+            threadPool,
+            actionFilters,
+            indexNameExpressionResolver,
+            SchemaUpdateRequest::new);
+        this.nodeClient = nodeClient;
+        this.xContentRegistry = xContentRegistry;
+    }
+
+    @Override
+    protected String executor() {
+        return ThreadPool.Names.GENERIC;
```

Comment:
afaik the master operation will do async very soon. wouldn't it be ok to stay in the SAME (netty) pool here?